controllers/octaviaapi_controller.go

```diff
@@ -191,35 +188,32 @@ func (r *OctaviaAPIReconciler) SetupWithManager(mgr ctrl.Manager) error {
 func (r *OctaviaAPIReconciler) reconcileDelete(ctx context.Context, instance *octaviav1.OctaviaAPI, helper *helper.Helper) (ctrl.Result, error) {
 	util.LogForObject(helper, "Reconciling Service delete", instance)
 
-	// It's possible to get here before the endpoints have been set in the status, so check for this
-	if instance.Status.APIEndpoints != nil {
-		// Remove the finalizer from our KeystoneEndpoint CR
-		keystoneEndpoint, err := keystonev1.GetKeystoneEndpointWithName(ctx, helper, octavia.ServiceName, instance.Namespace)
-		if err != nil && !k8s_errors.IsNotFound(err) {
+	// Remove the finalizer from our KeystoneEndpoint CR
+	keystoneEndpoint, err := keystonev1.GetKeystoneEndpointWithName(ctx, helper, octavia.ServiceName, instance.Namespace)
+	if err != nil && !k8s_errors.IsNotFound(err) {
+		return ctrl.Result{}, err
+	}
+
+	if err == nil {
+		controllerutil.RemoveFinalizer(keystoneEndpoint, helper.GetFinalizer())
```

Comment:
Maybe we could look at the boolean return value from `controllerutil.RemoveFinalizer()` and then only call the following update if the finalizer was in fact removed?  It seems unnecessary to always call `Update()` if in some cases the finalizer might already be gone.

Reply:
Something like this: https://github.com/openstack-k8s-operators/keystone-operator/blob/2e5fe448c4990abddbd7ba4ee7463c6b0e2e5418/controllers/keystoneendpoint_controller.go#L275-L281

Reply:
Yes I think that makes sense. I'll update the other operators as well.